### ISSUE TITLE
What is the difference between your method and the outlines' ?

### ISSUE THREAD

**Issue by AL-377** [chenhongqiao/ToolDec#2]
There is a tool called [outlines](https://github.com/outlines-dev), and it guides LLM generation in the same way(FSM and logits modify).So what is the difference between your methods?

**Comment by lileicc** [comment]
thanks for mentioning that! we were not aware of outlines before. We will check out the details.

**Comment by chenhongqiao** [comment]
Thank you for pointing out the repo. It's great to see other work in the same domain. While there are similarities, our approach and experiments offer several unique contributions:

- Our method allows the switching between "tool mode" and "text mode". This enables the generation of interleaved text and tool calls to effectively tackle problems, which is distinct from their "guided generation" that focuses mainly on generating structural data like JSON and regex.
- We dive deeper into evaluating the impact of descriptive tool names and signatures on the model's generalization ability. This exploration provides valuable insights into the model's generalization performance.
- In addition to addressing syntax errors, our research provides comprehensive experiments that demonstrate improved problem-solving and enhanced generalization when syntax errors are minimized. This showcases the practical significance of our findings.

Our work complements their concurrent work, and we believe it adds valuable insights to the field. We are happy to look into this work further. Please let us know if you have any other questions!

**Comment by AL-377** [comment]
Pretty amazing work，looking forward to the code!
